File: Java_projet_perso_/src/main/java/Trapeze.java
import java.io.BufferedReader;
import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.io.PrintWriter;
import java.util.concurrent.TimeUnit;

public class Trapeze extends Traitement{
	public float pointA;
	public float pointB;
	public float pas;
	public float resultat;
	
	public void setFonction(Fonction fx) {
		this.f = fx;
	}

	public void setPointA(float i) {
		this.pointA = i;
	}

	public void setPointB(float i) {
		this.pointB = i;
	}

	public void setPas(float i) {
		this.pas = i;
	}	

	public Trapeze() {
		
	}

	public void resolution(String path) throws FileNotFoundException {
		try {
			this.methode="Trapeze";
			PrintWriter fd = new PrintWriter(new File(path));
			float r=0;
			float a=this.pointA,b=this.pointB;
			float x=0,y=0;
			float h=(Math.abs(b)-Math.abs(a))/this.pas;
			
			for(int i=0;i<this.pas;i++){
				fd.println(a+" "+f.f(a));
				if(f.f(a)>f.f(a+h)){y=a;x=a+h;}
				else if(f.f(a)<=f.f(a+h)){y=a+h;x=a;}
				r+=((f.f(x)+f.f(y))*h)/2;
				a+=h;
			}
			fd.close();
			this.pathDatas=path;
			this.resultat=r;		
		}catch(Exception e) {
			e.getStackTrace();
			this.pathDatas="null";
		}
	}
}

File: php_projet_perso_codeigniter/app/Libraries/Trapeze.php
<?php
namespace App\Libraries;

class Trapeze extends Traitement {
    public $pointA;
    public $pointB;
    public $pas;
    public $resultat;
  
    public function __construct() {
      
    }  
    
    public function setPointA(float $i) {
      $this->pointA = $i;
    }
  
    public function setPointB(float $i) {
      $this->pointB = $i;
    }
  
    public function setPas(float $i) {
      $this->pas = $i;
    }
  
    public function resolution(string $path) {
      $this->methode = "Trapeze";
      try {
          $outputFile = fopen($path, "w");
          if (!$outputFile) {
            throw new Exception("Failed to open file: " . $path);
          }
        $r = 0;
        $a = min($this->pointA, $this->pointB);
        $b = max($this->pointA, $this->pointB);
        $h =(float) (($b - $a) / $this->pas);
  
        for ($i = 0; $i < $this->pas; $i++) {
          fwrite($outputFile, $a . " " . $this->f->f($a) . PHP_EOL);
          $y = $a;
          $x = $a + $h;
          if ($this->f->f($a) > $this->f->f($a + $h)) {
            $y = $a;
            $x = $a + $h;
          }
          $r +=(float) ((($this->f->f($x) + $this->f->f($y)) * $h) / 2);
          $a += $h;
        }
  
        fclose($outputFile);
        $this->resultat = $r;
        $this->pathDatas = $path;
      } catch (Exception $e) {
        echo $e->getMessage() . PHP_EOL;
        $this->pathDatas = "null";
      }
    }
  }
?>